kalitectl.py

```diff
@@ -592,6 +605,85 @@ def status():
 }
 
 
+def diagnose():
+    """
+    This command diagnoses an installation of KA Lite
+    
+    It has to be able to work with instances of KA Lite that users do not
+    actually own, however it's assumed that the path and the 'kalite' commands
+    are configured and work.
+    
+    The function is currently non-robust, meaning that not all aspects of
+    diagnose data collection is guaranteed to succeed, thus the command could
+    potentially fail :(
+    
+    Example: KALITE_HOME=/home/otheruser/.kalite kalite diagnose --port=7007
+    """
+    
+    print("")
+    print("KA Lite diagnostics")
+    print("")
+    
+    # Tell users we are calculating, because checking the size of the
+    # content directory is slow. Flush immediately after.
+    print("Calculating diagnostics...")
+    sys.stdout.flush()
+    print("")
+    
+    # Key, value store for diagnostics
+    # Not using OrderedDict because of python 2.6
+    diagnostics = []
+    
+    diag = lambda x, y: diagnostics.append((x, y))
+    
+    diag("KA Lite version", kalite.__version__)
+    diag("python", sys.version)
+    diag("platform", "\n".join(os.uname()))
+    
+    try:
+        __, __, port = get_pid()
+        for addr in get_ip_addresses():
+            diag("server address", "http://%s:%s/" % (addr, port))
+        status_code = STATUS_RUNNING
+    except NotRunning as e:
+        status_code = e.status_code
+    
+    diag("server status", status.codes[status_code])
+    
+    settings_imported = True  # Diagnostics from settings
+    try:
+        from django.conf import settings
+        from django.template.defaultfilters import filesizeformat
+    except:
+        settings_imported = False
+        diag("Settings failure", traceback.format_exc())
+    
+    if settings_imported:
+        diag("installed in", os.path.dirname(kalite.__file__))
+        diag("content root", settings.CONTENT_ROOT)
+        diag("content size", filesizeformat(get_size(settings.CONTENT_ROOT)))
+        diag("user database", settings.DATABASES['default']['NAME'])
+        diag("assessment database", settings.DATABASES['assessment_items']['NAME'])
+        try:
+            from securesync.models import Device
+            device = Device.get_own_device()
+            diag("device name", str(device.name))
+            diag("device UUID", str(device.get_uuid()))
```

Comment:
This should just be `device.id`, rather than recomputing the id from first principles.